src/components/Items/Scorecard.vue

```diff
@@ -0,0 +1,282 @@
+        <canvas id="confetticanvas" class="fixed z-50"></canvas>
+
+        <!-- circular progress bar -->
+        <CircularProgress
+          v-if="isCircularProgressShown"
+          class="relative mx-auto"
+          :radius="circularProgressRadius"
+          :progressBarPercent="localProgressBarPercentage"
+          :stroke="circularProgressStroke"
+          :result="progressBarResult"
+          :key="reRenderKey"
+        >
+        </CircularProgress>
+
+        <!-- metric boxes -->
+        <div class="flex bp-500:flex-row flex-col justify-center gap-4 px-10">
+          <div
+            v-for="metric in metrics"
+            class="rounded-md bp-500:rounded-2xl bg-yellow-400 grid grid-cols-2 bp-500:grid-rows-2 bp-500:grid-cols-none lg:grid-cols-2 lg:grid-rows-none border-2 px-4 lg:px-6 w-full h-14 bp-500:h-20"
+            :key="metric"
+          >
+            <div
+              class="w-full h-full flex flex-row justify-center gap-2 bp-500:mt-2 lg:mt-0"
+            >
+              <!-- metric icon -->
+              <inline-svg
+                :src="getImageSource(metric.icon.source)"
+                class="h-6 bp-360:h-10 bp-500:h-6 lg:h-10 w-6 bp-360:w-10 bp-500:w-4 md:w-6 lg:w-10 place-self-center"
+                :class="metric.icon.class"
+              ></inline-svg>
+              <!-- numeric value of the metric -->
+              <p
+                class="text-xl bp-360:text-2xl md:text-3xl lg:text-4xl font-bold my-auto"
+              >
+                {{ metric.value }}
+              </p>
+            </div>
+            <!-- name of the metric -->
+            <div
+              class="text-center text-sm bp-320:text-base md:text-base lg:text-xl font-medium my-auto bp-500:whitespace-nowrap lg:whitespace-normal"
+            >
+              {{ metric.name }}
+            </div>
+          </div>
+        </div>
+
+        <!-- action buttons -->
+        <div class="place-self-center mt-3">
+          <!-- watch again button -->
+          <icon-button
+            :titleConfig="watchAgainButtonTitleConfig"
+            :buttonClass="watchAgainButtonClass"
+            @click="restartVideo"
+            data-test="watchAgainButton"
+          ></icon-button>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import CircularProgress from "@/components/UI/Progress/CircularProgress.vue";
+import Utilties from "@/services/Functional/Utilities.js";
+import IconButton from "@/components/UI/Buttons/IconButton.vue";
+
+const confetti = require("canvas-confetti");
+const confettiCanvas = document.getElementById("confetticanvas");
+const confettiHandler = confetti.create(confettiCanvas, {
+  resize: true,
+  useWorker: true,
+});
+const confettiDuration = 10; // duration (in seconds) for which the confetti animation should run
+var confettiColors = ["#ff718d", "#fdff6a"]; // colors for confetti particles
+const waitTime = 500; // a time delay to be used in the code
```

Comment:
a more specific name + comment would be better
```suggestion
const progressBarAnimationWaitTime = 500; // a time delay to be used for animating the progress bar

```